### PR TITLE
Card Container

### PR DESCRIPTION
**Pull request by alaa-sahloub**
Relates #36

###Create index.js file in the CardContainer component folder which do the following :

- Import React and PropTypes.
- Import Typography from ant design.
- Import the Card Components.
- Import style.css.
- Declare Title and Text.
- Create the cardContainer function with title and data props.
- Render title and the container which contain the cards in the li tag and if there is no data will handle message there is no data.
- Create default props and validate them.

###Create style.css file which contains styles for the cards div container using flex-box and other styles.

### CODE REVIEW

File: client/src/Components/CardContainer/index.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import PropTypes from 'prop-types';
+import { Typography } from 'antd';
+import Card from '../Common/Card';
+import './style.css';
+
+const { Title, Text } = Typography;
+const CardContainer = ({ title, data }) => (
+  <div>
+    <Title level={3} id="container-title">
+      {title}
```
Comment (Mu7ammadAbed) [inline client/src/Components/CardContainer/index.js:11]: Is there any case in the Figma that require the `title` to be different from `أفضل رياض الأطفال تقييمًا`?

File: client/src/Components/CardContainer/index.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    <div className="container-cards">
+      <ul>
+        {data.length !== 0 ? (
+          data.map((item) => (
+            <li>
+              <Card {...item} />
+            </li>
+          ))
+        ) : (
+          <Text>لم يتم إضافة رياض أطفال.</Text>
```
Comment (Mu7ammadAbed) [inline client/src/Components/CardContainer/index.js:22]: Use the `Empty` component from `antd` 
```suggestion
          <Empty description="لم يتم إضافة أي رياض أطفال" />
```

File: client/src/Components/CardContainer/index.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    {
+      title: ' ',
+      rating: ' ',
+      reviewersNo: ' ',
+      location: ' ',
+      minPrice: ' ',
+      maxPrice: ' ',
+      cover: ' ',
+    },
+  ],
```
Comment (Mu7ammadAbed) [inline client/src/Components/CardContainer/index.js:40]: If you call this component like this `<CardContainer />` a default prop with empty values will be rendered not the `Empty` component
```suggestion
  data: []
```

File: client/src/Components/CardContainer/index.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+CardContainer.propTypes = {
+  title: PropTypes.string,
+  data: PropTypes.arrayOf(
+    PropTypes.shape({
+      title: PropTypes.string,
+      rating: PropTypes.string,
+      reviewersNo: PropTypes.string,
+      location: PropTypes.string,
+      minPrice: PropTypes.string,
+      maxPrice: PropTypes.string,
```
Comment (Mu7ammadAbed) [inline client/src/Components/CardContainer/index.js:51]: will these values be a number or string?

File: client/src/Components/CardContainer/index.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    },
+  ],
+};
+CardContainer.propTypes = {
+  title: PropTypes.string,
+  data: PropTypes.arrayOf(
+    PropTypes.shape({
+      title: PropTypes.string,
+      rating: PropTypes.string,
+      reviewersNo: PropTypes.string,
```
Comment (Mu7ammadAbed) [inline client/src/Components/CardContainer/index.js:48]: number or string ?

**Review by Mu7ammadAbed — CHANGES_REQUESTED** [review]

File: client/src/Components/CardContainer/index.js [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+const CardContainer = ({ title, data }) => (
+  <div>
+    <Title level={3} id="container-title">
+      {title}
+    </Title>
+    <div className="container-cards">
+      <ul>
+        {data.length !== 0 ? (
+          data.map((item) => (
+            <li>
```
Comment (Mu7ammadAbed) [inline client/src/Components/CardContainer/index.js:17]:
```suggestion
            <li key={id} >
```

**Review by Mu7ammadAbed — CHANGES_REQUESTED** [review]

File: client/src/Components/CardContainer/style.css [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+.container-cards > ul > li{
+    margin: 0  20px 30px;
+    list-style-type: none;
+}
+#container-title{
+    text-align: center;
+    padding-top: 10px;
+    padding-bottom: 5px;
+    color: var(--main-color);
+}
```
Comment (Osama-you) [inline client/src/Components/CardContainer/style.css:20]: add a new line at the end

File: client/src/Components/CardContainer/index.js (hunk 6 above)
Comment (Israa91) [reply client/src/Components/CardContainer/index.js:17]: Here I think you should take the kindergartenId from the props (data) and put it as a key for the li

File: client/src/Components/CardContainer/index.js (hunk 1 above)
Comment (alaa-sahloub) [reply client/src/Components/CardContainer/index.js:11]: Thank you Muhammed for the review. Yes we have the different title in the search result
image

File: client/src/Components/CardContainer/index.js [hunk 8]
```diff
@@ -1,54 +1,51 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import { Typography } from 'antd';
+
 import Card from '../Common/Card';
 import './style.css';
 
-const { Title, Text } = Typography;
-const CardContainer = ({ title, data }) => (
+const { Title, Empty } = Typography;
```
Comment (Mu7ammadAbed) [inline client/src/Components/CardContainer/index.js:8]: `Import { Empty } from 'antd`

File: client/src/Components/CardContainer/index.js [hunk 9]
```diff
@@ -1,54 +1,51 @@
...
+
+const CardContainer = ({ title, data, searchText }) => (
   <div>
     <Title level={3} id="container-title">
       {title}
+      {searchText && <span className="search-text">{searchText}</span>}
```
Comment (Mu7ammadAbed) [inline client/src/Components/CardContainer/index.js:14]: Change the span

**Review by Mu7ammadAbed — CHANGES_REQUESTED** [review]

File: client/src/Components/CardContainer/index.js [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+    <div className="container-cards">
+      <ul>
+        {data.length !== 0 ? (
+          data.map((item) => (
+            <li key={item.id}>
+              <Card {...item} />
+            </li>
+          ))
+        ) : (
+          <Empty description="لم يتم إضافة أي رياض أطفال" />
```
Comment (Mu7ammadAbed) [inline client/src/Components/CardContainer/index.js:24]:
```suggestion
          <Empty description="لا يوجد رياض أطفال" />
```

**Review by Mu7ammadAbed — APPROVED** [review]

**Review by Alaalser — APPROVED** [review]

**Review by Israa91 — APPROVED** [review]

**Review by Osama-you — APPROVED** [review]
good job

**Pull request merged**